### PR TITLE
add disko-install command

### PR DESCRIPTION
**Pull request by Mic92**
disko-install combines disko and nixos-install into a single command.

Example usage:

  sudo ./disko-install --flake /someflake#eva --disk sda /dev/zvol/zroot/test

### CODE REVIEW

**Comment by Mic92** [comment]
It's also possible to just mount and install without repartitioning everything:

```console
$ sudo /disko-install --mode mount -d --flake /tmp/foo/etc/nixos#foo --disk vdb /dev/zvol/zroot/test
```

File: disko-install [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  -f, --flake FLAKE_URI#ATTR  Use the specified flake to install the NixOS configuration.
+  --dry-run                   Print the commands that would be run, but do not run them.
+  --show-trace                Show the stack trace on error.
+  -h, --help                  Show this help message.
+EOF
+}
+
+serialiaseArrayToNix() {
+  local -n array=$1
+  nixExpr="{ "
```
Comment (Lassulus) [inline disko-install:17]: Why not json?

Comment (Mic92) [reply disko-install:17]: Is this not the same complexity regarding escaping required on the serialisation part?

**Comment by Mic92** [comment]
Secrets is something I would add in another pull request.

**Comment by Mic92** [comment]
I also need to add flag to allow writing efivars.

File: modules.nix
Comment (Lassulus) [inline modules.nix:1]: what is this file for?

File: tests/disko-install/module.nix
Comment (Lassulus) [inline tests/disko-install/module.nix:1]: what is this file for?

**Pull request merged**